### ISSUE TITLE
Running multiple Puphpeteer instance in parallel

### ISSUE THREAD

**Issue by tugrulcan** [rialto-php/puphpeteer#125]
_This is a question more than a bug and could not be sure if that should be under this repo or under `rialto-php/rialto`_

Hello :wave: 
When we try to run another instance of puphpeteer, it waits for the previous process to finish to make a new connection. Is it possible to run multiple instances at the same? If so, how? If it's not possible atm, what part of this project needs improvement for that feature? I can try to help you to add that feature in that case.

**Comment by nesk** [comment]
Hi,

I'm not sure what you are trying to do, for me you want two instaciate the `Nesk\Puphpeteer\Puppeteer` class multiple times? This is perfectly possible and it works out of the box, check out this script:

```php
<?php

require_once 'vendor/autoload.php';

// First instance
$pu1 = new \Nesk\Puphpeteer\Puppeteer;
$br1 = $pu1->launch(['headless' => false]);

// Second instance
$pu2 = new \Nesk\Puphpeteer\Puppeteer;
$br2 = $pu2->launch(['headless' => false]);

echo shell_exec('ps | grep node');

sleep(10);
```

Try it, you should see the following output:

```
36819 ttys005    0:00.15 node /absolute/path/to/puphpeteer/node_modules/@nesk/rialto/src/node-process/serve.js /absolute/path/to/puphpeteer/src/PuppeteerConnectionDelegate.js {"idle_timeout":60,"log_node_console":false,"log_browser_console":false}
36860 ttys005    0:00.17 node /absolute/path/to/puphpeteer/node_modules/@nesk/rialto/src/node-process/serve.js /absolute/path/to/puphpeteer/src/PuppeteerConnectionDelegate.js {"idle_timeout":60,"log_node_console":false,"log_browser_console":false}
36878 ttys005    0:00.00 sh -c ps | grep node
36880 ttys005    0:00.00 grep node
```

We can see we have two node processes running at the same time. Also you should see two browsers opened on your OS (due to `['headless' => false]`).

**Comment by mokhosh** [comment]
@nesk I wanted to open another issue for this, but seemed a bit related, and it was also a question not an issue, so.

What do you think is the best approach performance-wise, if you're using phphpeteer to generate pdf exports? Isn't it better if we keep a node process running and just use that? Or should we boot a process and close it with each export?

**Comment by nesk** [comment]
It would be better to keep a process running. Starting a new Node process and a new browser each time you want to generate a PDF is not the best way to optimize things. 😉
